### PR TITLE
Simplify filter tables for list TNUMs

### PR DESCRIPTION
**Pull request by fingolfin**
This PR gets rid of half the entries of all list filter tables, by using that they are invariant under mutability; i.e. the filter tables entries for a mutable TNUM and its immutable variant are essentially identical (up to the IMMUTABLE flag).

This is achieved by first fixing three (clearly incorrect) instances were this invariant was violated; then adding a check to verify that the remaining filter table entries do satisfy it; and finally removing the IMMUTABLE variants of all table entries. Instead of them, the four functions in gap.c which read the data from the filter tables have been modified to emit variants for the immutable TNUMs.

This work was motivated by PR #1563, and serves as kind of a preparation for it; indeed, this PR here reduces the number of lines in `src` which contain the string `IMMUTABLE` from 1328 to 568, which should help a bit with that PR.

### CODE REVIEW

File: src/lists.c [hunk 1]
```diff
@@ -2807,6 +2807,38 @@ static Int CheckInit (
     /* check implications                                                  */
     for ( i = FIRST_LIST_TNUM;  i <= LAST_LIST_TNUM;  i++ ) {
 
+        if ( (i & IMMUTABLE) == 0 ) {
+            if ( ClearFiltsTNums[i]+IMMUTABLE != ClearFiltsTNums[i+IMMUTABLE]) {
+                Pr( "#W  ClearFiltsTNums [%s] mismatch between mutable and immutable\n",
+                    (Int)(InfoBags[i].name), 0 );
+                success = 0;
+            }
+            for ( j = 0;  j < ARRAY_SIZE(fnums);  j++ ) {
```
Comment (stevelinton) [inline src/lists.c]: Arguably -- these three loops should be three calls to a function.

**Comment by stevelinton** [comment]
Looks good to me.

File: src/lists.c (hunk 1 above)
Comment (fingolfin) [inline src/lists.c]: Should the remaining loops in `CheckInit` then also be functions?

Anyaway: strictly speaking, I bope these three loops would soon be removed again anyway, as part of PR #1563, as there'd be no more IMMUTABLE bit in the TNUM...

Comment (fingolfin) [inline src/lists.c]: To be clear: that was a serious open question, not sarcasm: Looking at `CheckInit`, it is very long and non-trivial to read, and could benefit from some refactoring, though I punted on that fo now, in order to not get distracted :-)

Comment (stevelinton) [inline src/lists.c]: Agreed. I just noticed the repetition between the three loops in the PR. I completely agree it can wait.

**Comment by markuspf** [comment]
Maybe you could mention in the commit message of `b100a4b1` that these entries were there twice.

**Comment by fingolfin** [comment]
I absolutely will, the commit message is not good (as so often, it's obvious to me now that you point it out, but wasn't when I wrote it)

**Comment by fingolfin** [comment]
I adjusted that commit message (and also tweaked a few of the others).

@stevelinton if it is OK with you, I'll not factor out that code in `CheckInit` now, because the plan is to remove it again sometime soon anyway.

**Comment by stevelinton** [comment]
Fine by me.

**Pull request merged**